src/swap.ts

```diff
@@ -0,0 +1,291 @@
+import { purry } from './purry';
+import { isString } from './isString';
+import { isArray } from './isArray';
+import { IterableContainer } from './_types';
+
+type isEqual<A, B> = A extends B ? (B extends A ? true : false) : false;
+
+type isLessThan<A extends number, B extends number> = isEqual<A, B> extends true
+  ? false
+  : 0 extends A
+  ? true
+  : 0 extends B
+  ? false
+  : isLessThan<Subtract<A, 1>, Subtract<B, 1>>;
+
+type BuildTuple<
+  L extends number,
+  T extends ReadonlyArray<unknown> = []
+> = T['length'] extends L ? T : BuildTuple<L, [...T, unknown]>;
+
+type Subtract<A extends number, B extends number> = BuildTuple<A> extends [
+  ...infer U,
+  ...BuildTuple<B>
+]
+  ? U['length']
+  : never;
+
+type IsNonNegativeInteger<T extends number> = number extends T
+  ? false
+  : `${T}` extends `-${string}` | `${string}.${string}`
+  ? false
+  : true;
+
+type Absolute<T extends number> = `${T}` extends `-${infer R}`
+  ? R extends `${infer N extends number}`
+    ? N
+    : never
+  : T;
+
+type StringToChars<T extends string> = string extends T
```

Comment:
Similar to my comment above, I think type's should describe the result, not what they do. I'd call this `CharactersTuple`